src/bloomberg/datalake/datalakequerydbconsumer/_data_models.py

```diff
@@ -174,6 +174,117 @@ def __hash__(self) -> int:
+
+
+class OutputColumn(Base):  # type: ignore
+
+    __tablename__ = "output_columns"
+
+    queryId = Column("queryId", String(100), primary_key=True)
+    catalogName = Column("catalogName", String(100), primary_key=True)
+    schemaName = Column("schemaName", String(100), primary_key=True)
+    tableName = Column("tableName", String(100), primary_key=True)
+    columnName = Column("columnName", String(100), primary_key=True)
+
+    __table_args__ = (
+        ForeignKeyConstraint([queryId], [QueryMetrics.queryId]),
+        {"extend_existing": True, "schema": "raw_metrics"},
+    )
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, OutputColumn):
+            return False
+        return cast(
+            bool,
+            self.queryId == other.queryId
+            and self.catalogName == other.catalogName
+            and self.schemaName == other.schemaName
+            and self.tableName == other.tableName
+            and self.columnName == other.columnName,
+        )
+
+    def __hash__(self) -> int:
+        return hash(frozenset([self.queryId, self.catalogName, self.schemaName, self.tableName, self.columnName]))
+
+
+class OutputColumnSource(Base):  # type: ignore
+
+    __tablename__ = "output_column_sources"
+
+    id = Column("id", BigInteger, autoincrement=True, primary_key=True)
+    queryId = Column("queryId", String(100), primary_key=True)
+    catalogName = Column("catalogName", String(100), primary_key=True)
+    schemaName = Column("schemaName", String(100), primary_key=True)
+    tableName = Column("tableName", String(100), primary_key=True)
+    columnName = Column("columnName", String(100), primary_key=True)
+    sourceCatalogName = Column("sourceCatalogName", String(100))
+    sourceSchemaName = Column("sourceSchemaName", String(100))
+    sourceTableName = Column("sourceTableName", String(100))
+
+    sourceColumnName = Column("sourceColumnName", String(100))
+    __table_args__ = (
+        ForeignKeyConstraint(
+            [
+                queryId,
+                catalogName,
+                schemaName,
+                tableName,
+                columnName,
+            ],
+            [
+                OutputColumn.queryId,
+                OutputColumn.catalogName,
+                OutputColumn.schemaName,
+                OutputColumn.tableName,
+                OutputColumn.columnName,
+            ],
+        ),
+        {"extend_existing": True, "schema": "raw_metrics"},
+    )
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, OutputColumnSource):
+            return False
+        return cast(
+            bool,
+            self.queryId == other.queryId
+            and self.catalogName == other.catalogName
+            and self.schemaName == other.schemaName
+            and self.tableName == other.tableName
+            and self.columnName == other.columnName
+            and self.sourceCatalogName == other.sourceCatalogName
```

Comment:
pointing out that here this equality in not equivalent with the definition of 'equality' in the db via the primary keys. this is probably intended just making sure that it is on purpose

Reply:
good point, two equal objects on this condition would not cause a conflict in the database.
this is mainly because of limitations with MySQL, a primary key always creates an index, and an index on all these fields would be too big and table creation would fail.
i think we should keep this to 'enforce' that the tuple of these 9 fields should be unique